tools/unused_modules.py

```diff
@@ -0,0 +1,75 @@
+import os
+import re
+
+rolesPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'roles'))
+modulesPath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'module'))
+
+basePath = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'module')) + os.sep
+
+existingModules = {}
+missingModules = {}
+commentedModules = {}
+usedModules = []
+unusedButCommentedModules = []
+
+# a list of folders that could be within a module
+skip_list = ['src', 'data', 'config', 'tests', 'report']
```

Comment:
A better way than using a skip list would be to look for the CMakeLists.txt file in the directory and finding a NUClearModule call inside it.